### PR TITLE
Support for subdirectory multisites

### PR DESCRIPTION
**Pull request by mjangda**
Properly handle subdirectory sites on a multisite install. The original code for this plugin was written for WordPress.com which only supports domains and has many related assumptions that fail when running in a subdirectory (either as part of a multisite or on its own).

### CODE REVIEW

**Comment by david-binda** [comment]
Thanks for working on this @mjangda !

When testing the patch in VIP Quickstart, I have noticed that the `WPCOM_Concat_Utils::is_internal_url` is wrongly evaluating admin styles.

For instance : `/wp-includes/css/dashicons.min.css` are being concatenated on the blog with ID = 1, but not on blog ID 2 in subdirectory install, as the conditional on https://github.com/Automattic/nginx-http-concat/pull/28/files#diff-b510983752630cb415db4b4533552daeR13 is being evaluated as true and thus returns `false`.

I'm still digging into this and will look into possible solutions, but wanted to flag that here in order to share the findings.

File: tests/test-concat-utils.php [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+			'single_site__internal_absolute_url' => array(
+				'/wp-content/plugins/plugin/style.css',
+				'https://example.com',
+				true,
+			),
+
+			'subdir_site__internal_absolute_url' => array(
+				'/wp-content/plugins/plugin/style.css',
+				'https://example.com/mysite',
+				false,
```
Comment (david-binda) [inline tests/test-concat-utils.php]: @mjangda why we should be evaluating `/wp-content/plugins/plugin/style.css` as external in subdirectory site case?

Comment (mjangda) [inline tests/test-concat-utils.php]: I think the use case I was thinking of was WordPress being installed a subdirectory. Probably just a mistake though :)

**Comment by david-binda** [comment]
Note: the tests are failing "just" because of PHPCS, which is not working properly.

**Comment by david-binda** [comment]
@mjangda I'm done with this PR. I have successfully tested the updates on 1) a single site installation 2) MS domain setup 3) MS directory setup.

I have used following scenario for my tests:

1. I have disabled debug bar and query monitor
1. loaded the site with old version of this mu-plugin in place and captured the HTML source
1. loaded the site with this branch in place and diffed the HTML source against the previously captured one

The diff was clean (JS and CSS wise) in all cases mentioned above.

Could you, please, have a look at the changes I have done and eventually proceed with merging this PR?

**Comment by mjangda** [comment]
Haven't tested it yet, but the code looks solid. Will run through on my test site and follow-up.

**Comment by david-binda** [comment]
@mjangda I have nerfed the PHPCS checks for good and tests are now passing. How are the changes looking on your test site?

**Comment by mjangda** [comment]
Tested across a bunch of different configurations and looks pretty good. The only issue I spotted is that concat was broken for single site subdirectory installs (i.e. site installed and served from `example.com/subdir` or when installed in `/subdir` but served from `/`).

Pushed a fix for this 8396984 and re-tested and seems to be working well. If you agree, I can look at merging/deploying sometime today (Friday).

**Comment by david-binda** [comment]
@mjangda the patch looks good and it tests out during my testing.

Ship it!

**Comment by mjangda** [comment]
:boom:

**Pull request merged**